src/app/app-platform/ContentApp.h

```diff
@@ -94,11 +96,30 @@ class ContentAppClientCommandSender
 class DLL_EXPORT ContentApp
 {
 public:
+
+    struct SupportedCluster {
+        int clusterIdentifier { -1 };
```

Comment:
Is there a reason this is not using the ClusterId type and the relevant invalid value?

Because `int` cannot express all cluster ids....

Reply:
That is an interesting question. The original reason for this choice was to map the types exactly after the SupportedCluster defined here https://github.com/project-chip/connectedhomeip/blob/master/examples/tv-app/android/App/common-api/src/main/aidl/com/matter/tv/app/api/SupportedCluster.aidl which are converted to this C++ type to make the type conversions one-to-one. But I see that the CluserId is defined as uint32. The discrepency in the existing code is likely because Java does not have a separate type to represent unsigned integers. My understanding is then that all the Java ints *should be* interpreted as unsigned and not signed values, and that they can safely/correctly be cast to uint32_t if I change this type.

Thanks for bringing this up, will make this change.

Reply:
Updated!